src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,557 @@
+            static_assert(longCredentialType.size() == maxCredentialTypeLength);
+            pdomain::Credentials const longCredentials{
+                {alice[1], std::string(longCredentialType)}};
+
+            env(pdomain::setTx(alice[0], longCredentials));
+
+            // One account can create multiple domains
+            BEAST_EXPECT(env.ownerCount(alice[0]) == 2);
+
+            auto tx = env.tx()->getJson(JsonOptions::none);
+            BEAST_EXPECT(tx[jss::TransactionType] == "PermissionedDomainSet");
+            BEAST_EXPECT(tx["Account"] == alice[0].human());
+
+            bool findSeq = false;
+            for (auto const& [domain, object] :
+                 pdomain::getObjects(alice[0], env))
+            {
+                findSeq = object["Sequence"] == tx["Sequence"];
+                if (findSeq)
+                {
+                    BEAST_EXPECT(domain.isNonZero());
+                    BEAST_EXPECT(
+                        object["LedgerEntryType"] == "PermissionedDomain");
+                    BEAST_EXPECT(object["Owner"] == alice[0].human());
+                    BEAST_EXPECT(
+                        pdomain::credentialsFromJson(object, pubKey2Acc) ==
+                        longCredentials);
+                    break;
+                }
+            }
+            BEAST_EXPECT(findSeq);
+        }
+
+        // Create new from existing account with 10 credentials.
+        pdomain::Credentials const credentials10{
+            {alice[2], "credential1"},
+            {alice[3], "credential2"},
+            {alice[4], "credential3"},
+            {alice[5], "credential4"},
+            {alice[6], "credential5"},
+            {alice[7], "credential6"},
+            {alice[8], "credential7"},
+            {alice[9], "credential8"},
+            {alice[10], "credential9"},
+            {alice[11], "credential10"},
+        };
+        uint256 domain2;
+        {
+            BEAST_EXPECT(
+                credentials10.size() ==
+                maxPermissionedDomainCredentialsArraySize);
+            BEAST_EXPECT(
+                credentials10 != pdomain::sortCredentials(credentials10));
+            env(pdomain::setTx(alice[0], credentials10));
+            auto tx = env.tx()->getJson(JsonOptions::none);
+            domain2 = pdomain::getNewDomain(env.meta());
+            auto objects = pdomain::getObjects(alice[0], env);
+            auto object = objects[domain2];
+            BEAST_EXPECT(
+                pdomain::credentialsFromJson(object, pubKey2Acc) ==
+                pdomain::sortCredentials(credentials10));
+        }
+
+        // Update with 1 credential.
+        env(pdomain::setTx(alice[0], credentials1, domain2));
+        BEAST_EXPECT(
+            pdomain::credentialsFromJson(
+                pdomain::getObjects(alice[0], env)[domain2], pubKey2Acc) ==
+            credentials1);
+
+        // Update with 10 credentials.
+        env(pdomain::setTx(alice[0], credentials10, domain2));
```

Comment:
Please add either an update or a new domain where one of the credentials is issued by the domain owner

Reply:
added